### ISSUE TITLE
Scrape address UI issue

### ISSUE THREAD

**Issue by mitchellcash** [PaycoinFoundation/paycoin#210]
So far, in my small testing, scraping to an external address seems to be working fine but there are some display issues that might need thinking about.

In the Transactions tab we aren't really highlighting where the Stake has gone. It can look a bit confusing.

For example, in the Address column on the Transactions page we put the address from which the stake came from but don't show any info as to where the Stake has gone. This now becomes a possible hole in the transaction record.

It is possible however when you view the transaction details of a Stake to see the address where you sent the stake, however, this now becomes confusing to the user because the Address Page is showing one address where the Transaction Details are showing another address.

![](https://dl.dropboxusercontent.com/u/37902134/Paycoin%20Core/%23193_0.png)

**Comment by IngCr3at1on** [comment]
I believe #247 resolves this.

**Comment by IngCr3at1on** [comment]
This should be resolved, can anyone confirm?

**Comment by IngCr3at1on** [comment]
Alright @mitchellcash, #247 is actually merged this time so have another look I think you'll find this is resolved.

**Comment by IngCr3at1on** [comment]
@mitchellcash can you confirm if this is resolved?

**Comment by mitchellcash** [comment]
Hmm I'm not sure, I would need more testing with a Standard Staking address but a quick look at my MicroPrime address on my mainnet wallet on Mac OS X I see this from 4 days ago:

![](https://dl.dropboxusercontent.com/u/37902134/GitHub/PaycoinFoundation%3Apaycoin/%23210_0.png)

But now look at https://ledger.paycoin.com/transaction/a0b76a50e7d52f48156b4cae31d28c8b1f7d924f1306e0cda93718c01410d169

You will see this was an external stake to PWeDVb9VV1Wyn2dSy1sNgGDBJPkzrGkFkQ not a mint by stake to the same address.

**Comment by IngCr3at1on** [comment]
@mitchellcash an "external stake" occurs when the staking address is in a different wallet than the one that it staked to... Looks like I would have expected otherwise.

**Comment by mitchellcash** [comment]
Oh right. Good point, I wrote this on my laptop in bed maybe not the best time for testing... 

In regards to same wallet scrapes outside of the scraping address I would think it should at least show the address it staked to and not the staking address. 

I would say the general rule is to always display the address the stake went to. If that is the same address that made the stake so be it, it will display that. 

I will test this weekend the actual external scraping display.

**Comment by IngCr3at1on** [comment]
@mitchellcash aye makes sense, it was specifically external scraping that made me notice what was going on (not sure I have a wallet scraping to the same wallet lol). I'll see about tweaking this to accommodate local a little better as well.

As for scraping to the same address that was supposed to be disabled for sanity sake. Why scrape coins to the same address? We're purposefully splitting the transaction when we do the scrape which adds to the number of outputs and therefor complexity of the transaction. Stake outputs are never greater than 2, so with a scrape that's 3 outputs max; regardless I can't imagine why anyone would want a bunch of little tiny inputs instead of just letting the reward go to the/(one of the) stake outputs.

That's why I did #365.

**Comment by mitchellcash** [comment]
Ahh right, poor wording/confusion on my part. When I say scrape to the same address I just mean your standard stake with no scrape address that goes to the same address. Although behind the scenes it might create a new address it should still show the staking address in the Transactions table.

My head hurts already but maybe we should get together with a few of us on Google hangout and come to a consensus on some standardised rules for displaying scrape address transactions.

My initial thoughts are to agree with @dbroyhill and either let it be (who cares, what's the difference) or show an error message.

**Comment by IngCr3at1on** [comment]
@mitchellcash yeah I'm cool with adding an error but I think I've been clear on my opinion of living it as is lol

**Comment by IngCr3at1on** [comment]
@mitchellcash I believe this is resolved with #430
